### PR TITLE
MAT-4790: Non-Boolean Measure Observation Expected Results

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-4790](https://jira.cms.gov/browse/MAT-4790)

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/util/PopulationsMap.ts [hunk 1]
```diff
@@ -84,108 +84,263 @@ export function triggerPopChanges(
...
+            targetPopulation.populationValues[denomExIn] &&
+            Number(targetPopulation.populationValues[denomExIn].expected) >= 0
+              ? Number(targetPopulation.populationValues[denomExIn].expected)
+              : 0;
 
-      if (changedPopulationObservations && criteriaReferenceID) {
-        const measureObservationId = changedPopulationObservations.filter(
-          (observation) => observation.criteriaReference === criteriaReferenceID
-        )[0].id;
+        const headLength =
```
Comment (gregory-akins) [inline src/util/PopulationsMap.ts]: Can you add some comments here, or refactor so it's a little more clear what is happening?

**Review by gregory-akins — DISMISSED** [review]

File: src/util/PopulationsMap.ts (hunk 1 above)
Comment (ethankaplan) [reply src/util/PopulationsMap.ts]: Done!

Comment (ethankaplan) [reply src/util/PopulationsMap.ts]: If I need to explain it deeper, let me know

File: src/util/PopulationsMap.test.ts [hunk 2]
```diff
@@ -1273,6 +1298,219 @@ it("Adding the observations on clicking the numerator exclusion and denominator
...
+    actual: false,
+    id: "pid-5",
+    criteriaReference: "",
+  };
+  const populationValues: DisplayPopulationValue[] = [];
+  populationValues.push(ipp);
+  populationValues.push(denom);
+  populationValues.push(denomExclu);
+  populationValues.push(numer);
+  populationValues.push(numerExclu);
```
Comment (jkotanchik-SB) [inline src/util/PopulationsMap.test.ts:1343]:
```suggestion
  const populationValues: DisplayPopulationValue[] = [ipp, denom, denomExclu, numer, numerExclu];
```
Pretty sure these are equivalent methods to set the initial array. Optional change.

File: src/util/PopulationsMap.test.ts [hunk 3]
```diff
@@ -1273,6 +1298,219 @@ it("Adding the observations on clicking the numerator exclusion and denominator
...
+  expect(
+    parsingTheExpectedResult(
+      resultPops4[0].populationValues,
+      PopulationType.MEASURE_OBSERVATION,
+      "Ratio"
+    ).id
+  ).toBeFalsy();
+});
+
+it("NonBool: Removing and Adding the observations on changine the measure population exclusion in continuous variable ", () => {
```
Comment (jkotanchik-SB) [inline src/util/PopulationsMap.test.ts]:
```suggestion
it("NonBool: Adding and Removing observations on changes to the measure population exclusion in continuous variable ", () => {
```

File: src/util/PopulationsMap.ts [hunk 4]
```diff
@@ -47,11 +47,11 @@ export const FHIR_POPULATION_CODES = {
 export function triggerPopChanges(
   groupPopulations: GroupPopulation[],
   changedGroupId: string,
+  //should the below be changedPopulation?
   changedStratification: DisplayPopulationValue,
```
Comment (jkotanchik-SB) [inline src/util/PopulationsMap.ts]: @sb-prateekkeerthi see inline comment. This function is called once and the passed in argument is `changedPopulation`

**Review by jkotanchik-SB — DISMISSED** [review]

**Comment by adongare** [comment]
@ethankaplan I pulled down your code in my local and noticed it is still not working as expected. Please see my screenshot. I entered 3 in the numerator but I see more than 3 boxes for observation. Maybe you are not considering the one that is already present. Please work with @nmorasb on this(or I can walk you through the whole process once I'm back which will delay this). 

Screen Shot 2022-10-07 at 2 39 41 PM

File: src/util/PopulationsMap.ts (hunk 4 above)
Comment (sb-prateekkeerthi) [reply src/util/PopulationsMap.ts]: @jkotanchik-SB  when I had first written the code I was using it only to track changed populations so I have named it changed populations but later stratification work is added and for some reason the property name is changed to changedStratification and this property is now being used to track both populations as well as stratifications, so I think may be now we need to use a generic name which would satisfy both changed Population and changed stratification

**Comment by ethankaplan** [comment]
> @ethankaplan I pulled down your code in my local and noticed it is still not working as expected. Please see my screenshot. I entered 3 in the numerator but I see more than 3 boxes for observation. Maybe you are not considering the one that is already present. Please work with @nmorasb on this(or I can walk you through the whole process once I'm back which will delay this).
> 
> 

That was a bug with the refactoring, we've reverted part of it and it works now!

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**